### PR TITLE
Error if license_file is empty

### PR DESCRIPTION
**Pull request by chrisburr**
Checklist
* [ ] Added a ``news`` entry

cc @beckermr

### CODE REVIEW

File: conda_smithy/lint_recipe.py [hunk 1]
```diff
@@ -290,10 +290,8 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
 
     # 12a: License family must be valid (conda-build checks for that)
     license_family = about_section.get("license_family", license).lower()
-    license_file = about_section.get("license_file", "")
-    if license_file == "" and any(
-        f for f in NEEDED_FAMILIES if f in license_family
-    ):
+    license_file = about_section.get("license_file", None)
+    if not license_file:
```
Comment (chrisburr) [inline conda_smithy/lint_recipe.py:294]: Should this use `NEEDED_FAMILIES` or should it require everything to have a licence file?

If `NEEDED_FAMILIES` is desirable:

```suggestion
    if not license_file and any(
        f for f in NEEDED_FAMILIES if f in license_family
     ):
```

**Comment by chrisburr** [comment]
@beckermr Done

**Comment by beckermr** [comment]
Frack - we missed the news entry here.

**Comment by beckermr** [comment]
please submit it in another PR if you have the chance

**Comment by chrisburr** [comment]
Will do :+1:

**Pull request merged**